src/test/scala/edu/ie3/simona/model/participant2/evcs/EvcsModelSpec.scala

```diff
@@ -0,0 +1,749 @@
+            Kilowatts(2.5)
+        )
+        nextEvent shouldBe Some(10800L)
+      }
+    }
+
+    "determining current state correctly" when {
+
+      "being provided with a ChargingSchedule consisting of one entry" in {
+        val evcsModel = createModel("constantpower")
+
+        val cases = Table(
+          (
+            "storedEnergy",
+            "lastStateTick",
+            "currentTick",
+            "power",
+            "expectedStored",
+          ),
+          // empty battery
+          (0.0, 900L, 2700L, 5.0, 2.5),
+          (0.0, 0L, 3600L, -5.0, 0.0),
+          // half full battery
+          (5.0, 0L, 3600L, 5.0, 10.0),
+          (5.0, 900L, 2700L, 5.0, 7.5),
+          (5.0, 0L, 3600L, -5.0, 0.0),
+          (5.0, 900L, 2700L, -5.0, 2.5),
+          // full battery
+          (10.0, 900L, 2700L, -5.0, 7.5),
+          (10.0, 0L, 3600L, 5.0, 10.0),
+        )
+
+        forAll(cases) {
+          (
+              storedEnergy,
+              lastStateTick,
+              currentTick,
+              power,
+              expectedStored,
+          ) =>
+            val ev = EvModelWrapper(
+              ev1.copyWith(storedEnergy.asKiloWattHour)
+            )
+
+            val state = EvcsState(
+              Seq(ev),
+              lastStateTick,
+            )
+
+            val operatingPoint = EvcsOperatingPoint(
+              Map(ev.uuid -> Kilowatts(power))
+            )
+
+            val newState = evcsModel.determineState(
+              state,
+              operatingPoint,
+              currentTick,
+              dateTime,
+            )
+
+            newState.evs should have size 1
+            newState.tick shouldBe currentTick
+
+            val actualEv = newState.evs.headOption.getOrElse(
+              fail("No charging schedule provided.")
+            )
+
+            actualEv.uuid shouldBe ev.uuid
+            actualEv.id shouldBe ev.id
+            actualEv.pRatedAc shouldBe ev.pRatedAc
+            actualEv.pRatedDc shouldBe ev.pRatedDc
+            actualEv.eStorage shouldBe ev.eStorage
+            actualEv.storedEnergy should approximate(
+              KilowattHours(expectedStored)
+            )
+            actualEv.departureTick shouldBe ev.departureTick
+
+        }
+
+      }
+    }
+
+    "calculate results correctly" when {
+
+      val evcsModel = createModel("constantpower")
+
+      val evA = EvModelWrapper(ev1)
+      val evB = EvModelWrapper(ev2)
+
+      "two EVs are parked and charging without last operating point" in {
+
+        val currentOperatingPoint = EvcsOperatingPoint(
+          Map(evA.uuid -> Kilowatts(3.0), evB.uuid -> Kilowatts(2.0))
+        )
+
+        val state = EvcsState(
+          Seq(evA, evB),
+          10800L,
+        )
+
+        val results = evcsModel.createResults(
+          state,
+          None,
+          currentOperatingPoint,
+          ComplexPower(Kilowatts(5), Kilovars(0.5)),
```

Comment:
With EVs, we basically don't have any reactive power... If this test doesn't try to check reactive power specifically, I would prefer to use only active power (or very little reactive power...).